src/wanderer/lib.ts

```diff
@@ -0,0 +1,248 @@
+  name: string;
+  value: number;
+  location: Location;
+  prepareTurn: () => boolean;
+
+  /**
+   * Process for determining where to put a wanderer to extract additional value from it
+   * @param name name of this wanderer - for documentation/logging purposes
+   * @param location returns the location to adventure to target this; null only if something goes wrong
+   * @param value the expected additional value of putting a single wanderer-fight into the zone for this
+   * @param prepareTurn attempt to set up, spending meat and or items as necessary
+   */
+  constructor(
+    name: string,
+    location: Location,
+    value: number,
+    prepareTurn: () => boolean = () => true
+  ) {
+    this.name = name;
+    this.value = value;
+    this.location = location;
+    this.prepareTurn = prepareTurn;
+  }
+}
+export type WandererFactory = (
+  type: DraggableFight,
+  locationSkiplist: Location[]
+) => WandererTarget[];
+export type WandererLocation = { location: Location; targets: WandererTarget[]; value: number };
+
+export const unsupportedChoices = new Map<Location, { [choice: number]: number | string }>([
+  [$location`The Spooky Forest`, { [502]: 2, [505]: 2 }],
+  [$location`Guano Junction`, { [1427]: 1 }],
+  [$location`The Hidden Apartment Building`, { [780]: 6, [1578]: 6 }],
+  [$location`The Black Forest`, { [923]: 1, [924]: 1 }],
+  [$location`LavaCo™ Lamp Factory`, { [1091]: 9 }],
+  [$location`The Haunted Laboratory`, { [884]: 6 }],
+  [$location`The Haunted Nursery`, { [885]: 6 }],
+  [$location`The Haunted Storage Room`, { [886]: 6 }],
+  [$location`The Hidden Park`, { [789]: 6 }],
+  [$location`A Mob of Zeppelin Protesters`, { [1432]: 1, [857]: 2 }],
+  [$location`A-Boo Peak`, { [1430]: 2 }],
+  [$location`Sloppy Seconds Diner`, { [919]: 6 }],
+  [$location`VYKEA`, { [1115]: 6 }],
+  [
+    $location`The Castle in the Clouds in the Sky (Basement)`,
+    {
+      [670]: 4,
+      [671]: 4,
+      [672]: 1,
+    },
+  ],
+  [
+    $location`The Haunted Bedroom`,
+    {
+      [876]: 1, // old leather wallet, 500 meat
+      [877]: 1, // old coin purse, 500 meat
+      [878]: 1, // 400-600 meat
+      [879]: 2, // grouchy spirit
+      [880]: 2, // a dumb 75 meat club
+    },
+  ],
+  [$location`The Copperhead Club`, { [855]: 4 }],
+  [$location`The Castle in the Clouds in the Sky (Top Floor)`, { [1431]: 1, [677]: 2 }],
+  [$location`The Hidden Office Building`, { [786]: 6 }],
+]);
+
+export function defaultFactory(): WandererTarget[] {
+  return [new WandererTarget("Default", $location`The Haunted Kitchen`, 0)];
+}
+
+export function maxBy<T>(array: T[], key: (t: T) => number): T {
+  return array
+    .map((t: T) => {
+      return { t, value: key(t) };
+    })
+    .reduce((prev, curr) => (prev.value < curr.value ? curr : prev)).t;
+}
+
+type WanderingSource = {
+  name: string;
+  item: Item;
+  max: number;
+  property: NumericProperty;
+  type: DraggableFight;
+};
+const WanderingSources: WanderingSource[] = [
```

Comment:
This information may or may not be in libram's wanderers code. I don't think it has been kept up to date. We should maybe port some of this over.